test/e2e/specs/editor/various/is-typing.spec.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'isTyping', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'should hide the toolbar when typing', async ( { editor, page } ) => {
+		// Enter to reach paragraph block.
+		await page.keyboard.press( 'Enter' );
+		// Insert paragraph
+		await page.keyboard.type( 'Type' );
+
+		const blockToolbar = page.locator(
+			'role=toolbar[name="Block tools"i]'
+		);
+
+		// Toolbar should not be showing
+		await expect( blockToolbar ).toBeHidden();
+
+		// Moving the mouse shows the toolbar.
+		await editor.showBlockToolbar();
+
+		// Toolbar is visible.
+		await expect( blockToolbar ).toBeVisible();
+
+		// Typing again hides the toolbar
+		await page.keyboard.type( ' and continue' );
+
+		// Toolbar is hidden again
+		await expect( blockToolbar ).toBeHidden();
+	} );
+
+	test( 'should not close the dropdown when typing in it', async ( {
+		editor,
+		page,
+	} ) => {
+		// Add a block with a dropdown in the toolbar that contains an input.
+		await editor.insertBlock( { name: 'core/query' } );
+
+		// Tab to Start Blank Button
+		await page.keyboard.press( 'Tab' );
+		// Select the Start Blank Button
+		await page.keyboard.press( 'Enter' );
+		// Select the First variation
+		await page.keyboard.press( 'Enter' );
+		// Moving the mouse shows the toolbar.
+		await editor.showBlockToolbar();
+		// Open the dropdown.
+		await page.getByLabel( 'Display settings' ).click();
```

Comment:
Nit: Could we change this to `getByRole`?

Reply:
Yep! `getByRole` is always preferred and TBH I haven't found a need to use other query functions yet. It's preferred because it communicates the intent well for readers, that we're clicking on a button rather than anything else.

Reply:
That makes sense. Thanks for explaining!